clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -394,7 +394,8 @@ else if (metadata.error() == Errors.TOPIC_AUTHORIZATION_FAILED)
         if (hasReliableLeaderEpoch && partitionMetadata.leaderEpoch.isPresent()) {
             int newEpoch = partitionMetadata.leaderEpoch.get();
             Integer currentEpoch = lastSeenLeaderEpochs.get(tp);
-            if (topicId != null && oldTopicId != null && !topicId.equals(oldTopicId)) {
+            // oldTopicId can be null (when metadata is fetched during topic recreation), update the metadata in that case as well.
```

Comment:
How about this?
```java
// Between the time that a topic is deleted and re-created, the client may lose
// track of the corresponding topicId (i.e. `oldTopicid` will be null). In this case, 
// when we discover the new topicId, we allow the corresponding leader epoch
// to override the last seen value.
```

Also, would it make sense to move this into the corresponding branch that it applies to?

Reply:
Ack. Changed the comment as per the suggestion.

> would it make sense to move this into the corresponding branch that it applies to?
Sorry, couldn't get it. Can you elaborate on this please.